api/v1alpha1/fenceagentsremediation_params.go

```diff
@@ -0,0 +1,416 @@
+	nodeNamesMap := make(map[string]bool)
+	for _, nodeMap := range spec.NodeParameters {
+		for nodeName := range nodeMap {
+			nodeNamesMap[string(nodeName)] = true
+		}
+	}
+	for nodeName := range spec.NodeSecretNames {
+		nodeNamesMap[string(nodeName)] = true
+	}
+
+	return slices.Collect(maps.Keys(nodeNamesMap))
+}
+
+// validateFenceAgentParams builds the fence agent parameters map with validation
+func validateFenceAgentParams(far *FenceAgentsRemediation, secretParams SecretParams) (map[ParameterName]string, error) {
+	nodeName := GetNodeName(far)
+	fenceAgentParams := make(map[ParameterName]string)
+
+	isNodeTemplateExistInSharedParams := false
+	// Validate and add shared parameters
+	for paramName, paramVal := range far.Spec.SharedParameters {
+		// Verify action must be reboot or off
+		if err := validateFenceAction(string(paramName), paramVal); err != nil {
+			return nil, err
+		}
+		// Verify param isn't already defined
+		if _, exist := fenceAgentParams[paramName]; exist {
+			err := fmt.Errorf(errorParamDefinedMultipleTimes, paramName)
+			paramsLog.Error(err, "can't build fence agents parameters a parameter is defined multiple times", "parameter name", paramName)
+			return nil, err
+		}
+
+		processedParamVal, err := template.RenderParameterTemplate(paramVal, nodeName)
+		if err != nil {
+			paramsLog.Error(err, "Failed to process template in shared parameter", "parameter", paramName, "value", paramVal, "node", nodeName)
+			return fenceAgentParams, err
+		}
+		isNodeTemplateExistInSharedParams = isNodeTemplateExistInSharedParams || paramVal != processedParamVal
+		fenceAgentParams[paramName] = processedParamVal
+	}
+
+	// Validate and add node parameters (these can override shared parameters)
+	for paramName, nodeMap := range far.Spec.NodeParameters {
+		if nodeVal, isFound := nodeMap[NodeName(nodeName)]; isFound {
+			// Verify action must be reboot or off
+			if err := validateFenceAction(string(paramName), nodeVal); err != nil {
+				return nil, err
+			}
+			// For node params we don't enforce uniqueness as node param value will override shared param
+			if _, exist := fenceAgentParams[paramName]; exist {
+				paramsLog.Info("Shared parameter is overridden by node parameter", "parameter", paramName)
+			}
+			fenceAgentParams[paramName] = nodeVal
+		} else {
+			paramsLog.Info("Node parameter is missing for this node", "parameter name", paramName, "node name", nodeName)
+		}
+	}
+
+	// Validate and add secret parameters
+	for secretKey, secretVal := range secretParams.params {
+		secretParam := ParameterName(secretKey)
+		// Verify action must be reboot or off
+		if err := validateFenceAction(string(secretParam), secretVal); err != nil {
+			return nil, err
+		}
+		if _, exist := fenceAgentParams[secretParam]; exist {
+			err := fmt.Errorf(errorParamDefinedMultipleTimes, secretParam)
+			paramsLog.Error(err, "can't build fence agents parameters a parameter is defined multiple times", "parameter name", secretParam)
```

Comment:
IMHO, a reasoning word is missing. Maybe store it as a var in the beginning, like `errorParamDefinedMultipleTimes`
```suggestion
			paramsLog.Error(err, "can't build fence agents parameters, when a parameter is defined multiple times", "parameter name", secretParam)
```